File: javascript/easy/findtheDifference.js
/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
var findTheDifference = function(s, t) {
    const arr = new Array(26);
    arr.fill(0);
    for(let c of s) {
        let index = c.charCodeAt(0) - 97;
        arr[index]++;
    }
    
    for(let c of t) {
        let index = c.charCodeAt(0) - 97;
        arr[index]--;
    }
    
    for(let i = 0; i < arr.length; i++) {
        if(arr[i] != 0) {
            return String.fromCharCode(97+i);
        }
    }
};

File: ruby/easy/find_the_difference.rb
# @param {String} s
# @param {String} t
# @return {Character}
def find_the_difference(s, t)
    map = Hash.new
    t.each_char do |c|
        map[c] = map[c] ? map[c]+1 : 1
    end
    s.each_char do |c|
        return c unless map[c]
        map[c] = map[c]-1
    end
    map.each do |k,v|
        return k if v == 1
    end
end
